registry/adapters/kubernetes/kubernetes.go

```diff
@@ -16,73 +16,179 @@ package kubernetes
 
 import (
 	"fmt"
+	"reflect"
 	"strings"
 	"sync"
 	"time"
 
-	log "github.com/Sirupsen/logrus"
+	"encoding/json"
 
 	"github.com/amalgam8/amalgam8/pkg/auth"
+	"github.com/amalgam8/amalgam8/pkg/datastructures"
+	"github.com/amalgam8/amalgam8/pkg/errors"
 	"github.com/amalgam8/amalgam8/registry/api"
 	"github.com/amalgam8/amalgam8/registry/utils/logging"
+	"k8s.io/client-go/kubernetes"
+	kubeapi "k8s.io/client-go/pkg/api"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/runtime"
+	"k8s.io/client-go/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
 )
 
 const (
-	module = "K8SADAPTER"
+	// EndpointsCacheResyncPeriod is the period in which we do a full resync of the endpoints cache.
+	EndpointsCacheResyncPeriod = time.Duration(60) * time.Second
+
+	// PodCacheResyncPeriod is the period in which we do a full resync of the endpoints cache.
```

Comment:
first one is used for the endpoints cache, and the second one for the pods cache. I'll fix the comments.